src/EditorFeatures/Core.Wpf/Interactive/InteractiveDocumentNavigationService.cs

```diff
@@ -58,35 +58,40 @@ public async Task<bool> TryNavigateToSpanAsync(Workspace workspace, DocumentId d
             var textSnapshot = document?.GetTextSynchronously(cancellationToken).FindCorrespondingEditorTextSnapshot();
             if (textSnapshot == null)
             {
-                return false;
+                return null;
             }
 
             var snapshotSpan = new SnapshotSpan(textSnapshot, textSpan.Start, textSpan.Length);
             var virtualSnapshotSpan = new VirtualSnapshotSpan(snapshotSpan);
 
             if (!textView.TryGetSurfaceBufferSpan(virtualSnapshotSpan, out var surfaceBufferSpan))
```

Comment:
Does this bit need to move into the deferred work? Since I think if there was async stuff streaming to the interactive window, the buffers could be updated and this would be stale?

Reply:
in this case, we're technically on the UI thread the entire time.  It's teh first line of GetLocationForSpanAsync.  So it's less clear how much value any split has here.  The split i was at least going for was having the first half be the part that detemines "is this safe to nav to?" and the second half is the actual "ok, now actually go there".